### PR TITLE
#13423 update InterimInvoiceFlatrateTerm.QtyOrdered when C_OrderLine.EnteredQty changes
Only allow invoking C_Flatrate_Term_CreateInterimContract process for a C_OrderLine that has no existing InterimInvoiceFlatrateTerm created for it

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/13423

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java [hunk 1]
```diff
@@ -45,6 +52,8 @@ public class C_Flatrate_Term_CreateInterimContract extends JavaProcess implement
 	@Param(mandatory = true, parameterName = "DateTo")
 	private Timestamp p_DateTo;
 
+	IInterimInvoiceFlatrateTermDAO interimInvoiceFlatrateTermDAO = Services.get(IInterimInvoiceFlatrateTermDAO.class);
```
Comment (metas-rc) [inline backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java:55]: final

File: backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java [hunk 2]
```diff
@@ -56,11 +65,21 @@ public ProcessPreconditionsResolution checkPreconditionsApplicable(final @NonNul
 		{
 			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection();
 		}
-		final I_C_OrderLine salesOrder = context.getSelectedModel(I_C_OrderLine.class);
-		if (!salesOrder.isProcessed())
+		final I_C_OrderLine orderLine = InterfaceWrapperHelper.load(context.getSingleSelectedRecordId(), I_C_OrderLine.class);
```
Comment (metas-rc) [inline backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java:68]: please, use de.metas.order.impl.AbstractOrderDAO#getOrderLineById(de.metas.order.OrderLineId)

File: backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java [hunk 3]
```diff
@@ -56,11 +65,21 @@ public ProcessPreconditionsResolution checkPreconditionsApplicable(final @NonNul
...
+		if (!orderLine.isProcessed())
 		{
 			return ProcessPreconditionsResolution.rejectWithInternalReason("only processed order lines are allowed");
 		}
+		final I_C_Order order = InterfaceWrapperHelper.load(orderLine.getC_Order_ID(), I_C_Order.class);
```
Comment (metas-rc) [inline backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java:73]: Please, use de.metas.order.impl.AbstractOrderDAO#getById(de.metas.order.OrderId)

File: backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java]: done

File: backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java]: done.

File: backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java (hunk 3 above)
Comment (adi-stefan) [reply backend/de.metas.contracts/src/main/java/de/metas/contacts/invoice/interim/process/C_Flatrate_Term_CreateInterimContract.java]: and done.

**Pull request merged**